src/components/AccountModule/connection-hooks.js

```diff
@@ -63,13 +63,26 @@ export function useWalletConnectionDetails(
 ) {
   const { walletNetworkName, hasNetworkMismatch } = useNetworkConnectionData()
   const theme = useTheme()
-  let connectionDetails = {}
   const isWalletAndClientSynced =
     Math.abs(walletSyncDelay - clientSyncDelay) <= OK_PROVIDER_SYNC_DELAY
   const networkSlowdown =
     walletSyncDelay >= MILD_PROVIDER_SYNC_DELAY &&
     clientSyncDelay >= MILD_PROVIDER_SYNC_DELAY &&
     isWalletAndClientSynced
+  const networkName = network.shortName.toLowerCase()
+
+  let connectionDetails = {
+    connectionMessage: `Connected to ${walletNetworkName}`,
+    connectionMessageLong: `Connected to Ethereum ${walletNetworkName} Network`,
+    connectionColor: theme.positive,
+  }
+
+  if (
+    clientListening &&
+    (networkName === 'local' || networkName === 'unknown')
+  ) {
+    return connectionDetails
+  }
```

Comment:
Maybe we can improve this entire construction of code by using:

1. A `const` declaration for the default, "working" connection state
2. Returns inside each if, since they're always changing the same keys

Reply:
Yep, definitely agree on doing this refactor, not a big fan of `let`. I'm so tempted to try and use XState later on this though haha